flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/context/SessionContextTest.java

```diff
@@ -79,78 +66,120 @@ public void testSetAndResetOption() throws Exception {
+
+        Assert.assertEquals("table", getMap(sessionContext).get("execution.result-mode"));
+
+        sessionContext.reset("execution.result-mode");
+        Assert.assertEquals("changelog", getMap(sessionContext).get("execution.result-mode"));
+        // no reset this key execution.max-table-result-rows
         Assert.assertEquals(
-                "test",
-                sessionContext
-                        .getExecutionContext()
-                        .getTableEnvironment()
-                        .getConfig()
-                        .getConfiguration()
-                        .getString(NAME));
-        Assert.assertFalse(
-                sessionContext
-                        .getExecutionContext()
-                        .getTableEnvironment()
-                        .getConfig()
-                        .getConfiguration()
-                        .getBoolean(OBJECT_REUSE));
+                "200000", getMap(sessionContext).get("execution.max-table-result-rows"));
 
-        sessionContext.reset();
+        // reset option for deprecated key
+        sessionContext.reset("sql-client.execution.max-table-result.rows");
         Assert.assertEquals(
-                "default",
-                sessionContext
-                        .getExecutionContext()
-                        .getTableEnvironment()
-                        .getConfig()
-                        .getConfiguration()
-                        .getString(TABLE_SQL_DIALECT));
+                "100", getMap(sessionContext).get("sql-client.execution.max-table-result.rows"));
+        Assert.assertEquals("100", getMap(sessionContext).get("execution.max-table-result-rows"));
+    }
+
+    @Test
+    public void testSetAndResetKeyInConfigOptions() throws Exception {
+        SessionContext sessionContext = createSessionContext();
+        // table config option
+        sessionContext.set(TABLE_SQL_DIALECT.key(), "hive");
+        // runtime config option and Yaml specified value
+        sessionContext.set(MAX_PARALLELISM.key(), "128");
+        // runtime config option and doesn't have default value
+        sessionContext.set(NAME.key(), "test");
+        // runtime config from flink-conf
+        sessionContext.set(OBJECT_REUSE.key(), "false");
+
+        Assert.assertEquals("hive", getConfiguration(sessionContext).getString(TABLE_SQL_DIALECT));
+        Assert.assertEquals(128, getConfiguration(sessionContext).getInteger(MAX_PARALLELISM));
+        Assert.assertEquals("test", getConfiguration(sessionContext).getString(NAME));
+        Assert.assertFalse(getConfiguration(sessionContext).getBoolean(OBJECT_REUSE));
+
+        sessionContext.reset(TABLE_SQL_DIALECT.key());
+        Assert.assertEquals(
+                "default", getConfiguration(sessionContext).getString(TABLE_SQL_DIALECT));
+
+        sessionContext.reset(MAX_PARALLELISM.key());
+        Assert.assertEquals(16, getConfiguration(sessionContext).getInteger(MAX_PARALLELISM));
```

Comment:
nit: would be better to import the static method and use `assertEquals` directly in the tests.